api/clients/eigenda_client.go

```diff
@@ -57,17 +63,44 @@ func NewEigenDAClient(log log.Logger, config EigenDAClientConfig) (*EigenDAClien
 }
 
 func (m EigenDAClient) GetBlob(ctx context.Context, BatchHeaderHash []byte, BlobIndex uint32) ([]byte, error) {
-	data, err := m.Client.RetrieveBlob(ctx, BatchHeaderHash, BlobIndex)
+	dataIFFT, err := m.Client.RetrieveBlob(ctx, BatchHeaderHash, BlobIndex)
 	if err != nil {
 		return nil, err
 	}
 
-	if len(data) == 0 {
+	if len(dataIFFT) == 0 {
 		return nil, fmt.Errorf("blob has length zero")
 	}
 
-	version := BlobEncodingVersion(data[0])
-	codec, err := BlobEncodingVersionToCodec(version)
+	// we assume the data is already IFFT'd so we FFT it before decoding
+	dataFrIFFT, err := rs.ToFrArray(dataIFFT)
+	if err != nil {
+		return nil, fmt.Errorf("error converting data to fr.Element: %w", err)
+	}
+	dataFrIFFTLen := uint64(len(dataFrIFFT))
+	dataFrIFFTLenPow2 := encoding.NextPowerOf2(dataFrIFFTLen)
+
+	if dataFrIFFTLenPow2 != dataFrIFFTLen {
+		return nil, fmt.Errorf("data length %d is not a power of 2", dataFrIFFTLen)
+	}
+
+	maxScale := uint8(math.Log2(float64(dataFrIFFTLenPow2)))
+
+	fs := fft.NewFFTSettings(maxScale)
+	dataFr, err := fs.FFT(dataFrIFFT, false)
+	if err != nil {
+		return nil, fmt.Errorf("failed to perform FFT: %w", err)
+	}
+
+	data := rs.ToByteArray(dataFr, dataFrIFFTLenPow2*encoding.BYTES_PER_SYMBOL)
+
+	// get version and length from codec blob header
+	version, _, err := codecs.DecodeCodecBlobHeader(data[:5])
```

Comment:
Don't we need to use the length here?

Reply:
don't need length, it's handled by the Decode function in the Codec this version byte points to